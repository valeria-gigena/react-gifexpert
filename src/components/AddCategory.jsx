import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmitForm = (event) => {
        event.preventDefault();
        if (inputValue.trim().length == 0) return;
        onNewCategory(inputValue);
        setInputValue('')
    }

    return (
        <form onSubmit={onSubmitForm} aria-label="form">
            <input
                type='text'
                onChange={(event) => onInputChange(event)}
                placeholder="Buscar en Giphy"
                value={inputValue}></input>
           
        </form>
    )
}

AddCategory.propTypes = { onNewCategory: PropTypes.func.isRequired }