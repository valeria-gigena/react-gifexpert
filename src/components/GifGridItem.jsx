import React from 'react'
import {getGifs} from '../helpers/getGifs'

const GifGridItem = ({ category }) => {
  getGifs(category);

  return (
    <>

      <h3>{category}</h3>

    </>
  )
}

export default GifGridItem
