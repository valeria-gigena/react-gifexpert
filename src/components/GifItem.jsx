import React from 'react'
import PropTypes from 'prop-types'

export const GifItem = ({ title, id, url }) => {
    return (
        <div className='card' key={id}>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}