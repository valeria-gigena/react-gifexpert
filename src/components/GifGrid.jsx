import React from 'react'
import GifGridItem from './GifGridItem'

const GifGrid = ({ categories }) => {
  return (

    <ol>
      {categories.map(item => {
        return <GifGridItem category={item}></GifGridItem>
        // return <li key={item}>{item}</li> 
      })


      }
    </ol>

  )
}

export default GifGrid
