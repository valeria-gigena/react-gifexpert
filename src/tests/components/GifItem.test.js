
import { render, screen } from "@testing-library/react";
import { GifItem } from '../../components/GifItem'

describe('Pruebas en <GifItem>', () => {
    const title = 'Saitama';
    const url = 'http://one-ponch/saitama.png';


    test('debe hacer match SnapShot ', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
    test('debe mostrar imagen con URL y ALT indicado ', () => {
        render(<GifItem title={title} url={url} />);
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect (src).toBe(url);
        expect (alt).toBe(title);
    });
    test('que se muestre correctamente el titulo',()=>{
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByText("0")).toBeInTheDocument();
        expect(screen.getByText(title)).toBeTruthy();
    })



})
