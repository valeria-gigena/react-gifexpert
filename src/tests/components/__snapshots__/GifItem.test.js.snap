// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem> debe hacer match SnapShot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one-ponch/saitama.png"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
